### PR TITLE
Feature store preview

### PR DESCRIPTION
**Pull request by KshitizLohia**
### Description
1. Creation of feature store entities as a stack
2. Spark data-frame in and spark data-frame out interface with spark magic 
3. Pandas data-frame in and pandas data-frame out interface. 
4. Lineage tree in the notebook using graphviz

## Documentation
https://feature-store-accelerated-data-science.readthedocs.io/en/latest/

### Integrations

- `great expectations`: Feature store supports great expectations validations for validation of data as this library seems to be mature, could be run with minimal additional dependencies and there’s a possibility to scale with apache spark.

- `mlm`: Feature store supports mlm profiler for statistics of data as this library could be run with minimal additional dependencies and there’s a possibility to scale with apache spark.

- `plotly`: Feature store uses plotly to showcase histograms and charts for each feature within a feature group or dataset.

- `delta-spark`: Feature store uses delta for each commit to allow time travel of queries and to allow capabilities like schema enforcement and schema evolution.

### Unit tests
Screenshot 2023-06-15 at 11 38 16 PM

### Integration tests
Screenshot 2023-06-29 at 2 29 46 PM

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -70,8 +70,11 @@
         "inflection",
     ],
     "bds": ["ibis-framework[impala]", "hdfs[kerberos]", "sqlalchemy"],
-    "spark": ["pyspark>=3.0.0"],
+    "spark": ["pyspark>=3.0.0", "delta-spark"],
```
Comment (mrDzurb) [inline setup.py:73]: These dependencies:
```
"spark": ["pyspark>=3.0.0", "delta-spark"],
"great-expectations": ["great-expectations==0.15.39"],
"mlm_insights": ["mlm_insights==0.1.0.dev1"],
"pyarrow": ["pyarrow"],
```
They are all feature store related, right? WOuldn't it be better to have something like this?

```
"feature-store": [
    "pyspark>=3.0.0", 
     "delta-spark",
     "great-expectations==0.15.39"
]
```

If so, The same change needs to be done in the runtime_dependency.

**Review by mrDzurb — DISMISSED** [review]

**Comment by lvallad** [comment]
@KshitizLohia Since I couldn't push my revisions, I zipped them into a file for you to do so.
[tc-review.zip](https://github.com/oracle/accelerated-data-science/files/13169159/tc-review.zip)

File: pyproject.toml
Comment (kumar-shivam-ranjan) [inline pyproject.toml:1]: Is there anywhere mentioned all the list of dependencies (great-expectations ,mlm, oci ,etc) required by feature-store module of ads(`oracle-ads[feature-store]`).  I currently have to install all the dependencies manually inside the virtual env.  If there is some requirements.txt or something , which could be helpful here?

**Comment by KshitizLohia** [comment]
> @KshitizLohia Since I couldn't push my revisions, I zipped them into a file for you to do so. [tc-review.zip](https://github.com/oracle/accelerated-data-science/files/13169159/tc-review.zip)

thanks for helping us with the documentation,  lorna 💯

**Review by liudmylaru — DISMISSED** [review]

**Review by mrDzurb — DISMISSED** [review]

**Review by liudmylaru — APPROVED** [review]

**Review by mrDzurb — APPROVED** [review]

**Pull request merged**